enterprise/licensing/src/main/java/io/crate/expression/reference/sys/check/cluster/LicenseCheck.java

```diff
@@ -72,34 +71,54 @@ public boolean validate() {
         LicenseService.LicenseState licenseState = licenseService.getLicenseState();
         switch (licenseState) {
             case VALID:
-                severity = LOW;
-                description = LICENSE_OK;
-                return true;
-
+                if (currentLicense.millisToExpiration() <= Duration.ofDays(1).toMillis()) {
+                    description = buildDescription(currentLicense);
+                    severity = HIGH;
+                    return false;
+                } else if (currentLicense.millisToExpiration() <= Duration.ofDays(15).toMillis()) {
+                    description = buildDescription(currentLicense);
+                    severity = MEDIUM;
+                    return false;
+                } else {
+                    description = LicenseExpiryNotification.LICENSE_VALID;
+                    severity = LOW;
+                    return true;
+                }
             case EXPIRED:
-                LicenseExpiryNotification licenseExpiryNotification = LicenseExpiryNotification.of(currentLicense);
-                description = getLinkedDescription(ID,
-                    licenseExpiryNotification.notificationMessage(currentLicense.millisToExpiration()),
-                    "For more information on Cluster Checks please visit: " + CLUSTER_CHECK_LINK_PATTERN);
-                severity = licenseExpiryNotification.equals(MODERATE) ? MEDIUM : HIGH;
+                description = buildDescription(currentLicense);
+                severity = HIGH;
                 return false;
 
             case MAX_NODES_VIOLATED:
+                description = buildMaxNodeViolatedDescription(licenseService.currentLicense(),
+                    clusterService.state().getNodes().getSize());
                 severity = HIGH;
-                description = getLinkedDescription(ID, String.format(
-                    Locale.ENGLISH,
-                    "The license is limited to %d nodes, but there are %d nodes in the cluster. " +
-                    "To upgrade your license visit https://crate.io/license-update/",
-                    licenseService.currentLicense().maxNumberOfNodes(),
-                    clusterService.state().getNodes().getSize()
-                ), "For more information visit: " + CLUSTER_CHECK_LINK_PATTERN);
                 return false;
 
             default:
                 throw new AssertionError("Illegal license state: " + licenseState);
         }
     }
 
+    private String buildDescription(@Nullable LicenseData license) {
+        assert license != null;
```

Comment:
just saw that its already ensured that a null licenseData won't end up here https://github.com/crate/crate/pull/8398/files#diff-387750345e9a66cd116da388108c8f22L70.
so we can revert nullable and assert, sorry!